tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -668,16 +668,18 @@ void TangoRosNode::OnPoseAvailable(const TangoPoseData* pose) {
         pose->frame.target == TANGO_COORDINATE_FRAME_DEVICE) {
       if (pose->status_code == TANGO_POSE_VALID &&
           device_pose_thread_.data_available_mutex.try_lock()) {
-        tango_ros_conversions_helper::toTransformStamped(*pose, time_offset_,
-                                                 &start_of_service_T_device_);
-        TangoCoordinateFramePair pair;
-        pair.base = TANGO_COORDINATE_FRAME_AREA_DESCRIPTION;
-        pair.target = TANGO_COORDINATE_FRAME_START_OF_SERVICE;
-        TangoPoseData area_description_T_start_of_service;
-        TangoService_getPoseAtTime(0.0, pair, &area_description_T_start_of_service);
-        if (area_description_T_start_of_service.status_code == TANGO_POSE_VALID) {
-          tango_ros_conversions_helper::toTransformStamped(area_description_T_start_of_service,
-                             time_offset_, &area_description_T_start_of_service_);
+        tango_ros_conversions_helper::toTransformStamped(
+            *pose, time_offset_, &start_of_service_T_device_);
+        if (tango_data_available_) {
```

Comment:
This solution couples the save map function with onPoseAvailable, which is quite unexpected. Either add a big comment why this is necessary or consider the following alternative solution which avoids a manual call to getPoseAtTime. Register multiple frame pairs so getPoseAtTime is called for sos_T_adf and sos_T_device.
```
void TangoRosNode::OnPoseAvailable(const TangoPoseData* pose) {
  if (base == area description && target == device) {
      // Save start_of_service_T_device
  } else if (base == area description && target == start of service
     // Save area_description_T_start_of_service_
  }
)
```